core/workflow-operator/src/main/scala/edu/uci/ics/amber/operator/state/DataToStateOpDesc.scala

```diff
@@ -0,0 +1,46 @@
+package edu.uci.ics.amber.operator.state
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaTitle
+import edu.uci.ics.amber.core.executor.OpExecWithClassName
+import edu.uci.ics.amber.core.virtualidentity.{ExecutionIdentity, WorkflowIdentity}
+import edu.uci.ics.amber.core.workflow.{InputPort, OutputPort, PhysicalOp}
+import edu.uci.ics.amber.operator.LogicalOp
+import edu.uci.ics.amber.operator.metadata.{OperatorGroupConstants, OperatorInfo}
+import edu.uci.ics.amber.util.JSONUtils.objectMapper
+
+class DataToStateOpDesc extends LogicalOp {
+  @JsonProperty(defaultValue = "false")
+  @JsonSchemaTitle("Pass To All Downstream")
+  @JsonDeserialize(contentAs = classOf[java.lang.Boolean])
+  var passToAllDownstream: Option[Boolean] = Option(false)
+
+  override def getPhysicalOp(
+      workflowId: WorkflowIdentity,
+      executionId: ExecutionIdentity
+  ): PhysicalOp = {
+    PhysicalOp
+      .oneToOnePhysicalOp(
+        workflowId,
+        executionId,
+        operatorIdentifier,
+        OpExecWithClassName(
+          "edu.uci.ics.amber.operator.state.DataToStateOpExec",
+          objectMapper.writeValueAsString(this)
+        )
+      )
+      .withInputPorts(operatorInfo.inputPorts)
+      .withOutputPorts(operatorInfo.outputPorts)
+      .withSuggestedWorkerNum(1)
```

Comment:
The suggested worker number is for user to give hint (so that it is called "suggested"). If you want to say that it requires a single worker, please set parallelable to false.

Is this parallel able? I think in general it is parallel able, but it needs to maintain the order of the input and output tuple. that might be the reason you are trying to use one worker to prevent from changing the order. This reveals a difference here: for all data tuples, they are "bag of tuples" thus order does not matter (with sort is a different case), but for all states, they need to be enforced on the order.